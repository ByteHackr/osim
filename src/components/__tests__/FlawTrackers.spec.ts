import { describe, expect } from 'vitest';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';

import FlawTrackers from '@/components/FlawTrackers.vue';

import { osimFullFlawTest, osimRequiredFlawTest } from './test-suite-helpers';

createTestingPinia();
vi.mock('@/composables/useTrackers', () => ({
  useTrackers: vi.fn().mockReturnValue({
    trackersToFile: [],
  }),
}));

describe('flawTrackers', () => {
  let subject;

  osimRequiredFlawTest('Correctly renders the component when there are not trackers to display', async ({ flaw }) => {
    subject = mount(FlawTrackers, {
      props: {
        flawId: flaw.uuid,
        displayedTrackers: [],
        affectsNotBeingDeleted: [],
        allTrackersCount: 0,
      },
    });
    expect(subject.html()).toMatchSnapshot();
  });

  osimFullFlawTest('Correctly renders the component when there are trackers to display', async ({ flaw }) => {
    subject = mount(FlawTrackers, {
      props: {
        flawId: flaw.uuid,
        displayedTrackers: flaw.affects
          .flatMap(affect => affect.trackers
            .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
          ),
        affectsNotBeingDeleted: [],
        allTrackersCount: 0,
      },
    });
    expect(subject.html()).toMatchSnapshot();
  });

  osimFullFlawTest('Per page setting correctly changes the table page items number', async ({ flaw }) => {
    subject = mount(FlawTrackers, {
      props: {
        flawId: flaw.uuid,
        displayedTrackers: flaw.affects
          .flatMap(affect => affect.trackers
            .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
          ),
        affectsNotBeingDeleted: [],
        allTrackersCount: 0,
      },
    });

    let trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    let rowCount = trackersTableRows.length;
    expect(rowCount).toBe(6);
    let itemsPerPageIndicator = subject.find('.trackers-toolbar .tracker-badges .btn span');
    expect(itemsPerPageIndicator.text()).toBe('Per page: 10');

    const reduceItemsBtn = subject.find('.trackers-toolbar .tracker-badges .btn .bi-dash-square');
    for (let i = 0; i < 5; i++) {
      await reduceItemsBtn.trigger('click');
    }

    trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    rowCount = trackersTableRows.length;
    expect(rowCount).toBe(5);
    itemsPerPageIndicator = subject.find('.trackers-toolbar .tracker-badges .btn span');
    expect(itemsPerPageIndicator.text()).toBe('Per page: 5');

    const increaseItemsBtn = subject.find('.trackers-toolbar .tracker-badges .btn .bi-plus-square');
    for (let i = 0; i < 5; i++) {
      await increaseItemsBtn.trigger('click');
    }

    trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    rowCount = trackersTableRows.length;
    expect(rowCount).toBe(6);
    itemsPerPageIndicator = subject.find('.trackers-toolbar .tracker-badges .btn span');
    expect(itemsPerPageIndicator.text()).toBe('Per page: 10');
  });

  osimFullFlawTest('Trackers can be sorted by clicking on the date field columns', async ({ flaw }) => {
    subject = mount(FlawTrackers, {
      props: {
        flawId: flaw.uuid,
        displayedTrackers: flaw.affects
          .flatMap(affect => affect.trackers
            .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
          ),
        affectsNotBeingDeleted: [],
        allTrackersCount: 0,
      },
    });

    let trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    let firstTracker = trackersTableRows[0];
    expect(firstTracker.find('td:nth-of-type(3)').text()).toBe('xxxx-0-006');

    const componentHeader = subject.find('.affects-trackers table thead tr th:nth-of-type(6)');
    expect(componentHeader.exists()).toBe(true);
    expect(componentHeader.text()).toBe('Created date');
    await componentHeader.trigger('click');

    trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    firstTracker = trackersTableRows[0];
    expect(firstTracker.find('td:nth-of-type(2)').text()).toBe('xxxx-0-001');
  });

  osimFullFlawTest('Displays embedded trackers manager', async ({ flaw }) => {
    subject = mount(FlawTrackers, {
      props: {
        flawId: flaw.uuid,
        displayedTrackers: flaw.affects
          .flatMap(affect => affect.trackers
            .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
          ),
        affectsNotBeingDeleted: [],
        allTrackersCount: 0,
      },
    });

    let trackerManagerElement = subject.find('.trackers-manager');
    expect(trackerManagerElement.exists()).toBe(false);

    let toggleTrackerManagerView = subject.find('.trackers-toolbar .btn-info');
    expect(toggleTrackerManagerView.text()).toBe('Show Trackers Manager');
    await toggleTrackerManagerView.trigger('click');

    trackerManagerElement = subject.find('.trackers-manager');
    expect(trackerManagerElement.exists()).toBe(true);

    toggleTrackerManagerView = subject.find('.trackers-toolbar .btn-info');
    expect(toggleTrackerManagerView.text()).toBe('Hide Trackers Manager');

    expect(subject.html()).toMatchSnapshot();
  });
});
