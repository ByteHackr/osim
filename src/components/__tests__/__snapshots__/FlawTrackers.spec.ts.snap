// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`flawTrackers > Correctly renders the component when there are not trackers to display 1`] = `
"<div data-v-89a61d08="">
  <div data-v-89a61d08="" class="affects-trackers">
    <div data-v-89a61d08="" class="bg-dark rounded-top-2 text-info">
      <div data-v-89a61d08="" class="affect-trackers-heading d-flex p-1 pt-2 px-3">
        <h5 data-v-89a61d08="" class="m-0">Trackers</h5>
      </div>
      <!--v-if--><input data-v-89a61d08="" type="text" class="form-control border border-info focus-ring m-2 mb-1 p-1" style="width: 35ch;" placeholder="Filter Modules/Components...">
      <div data-v-89a61d08="" class="trackers-toolbar p-2 pt-1">
        <div data-v-89a61d08="" class="tracker-badges">
          <!--v-if-->
          <!--v-if-->
        </div><button data-v-89a61d08="" type="button" class="btn btn-sm btn-info ms-auto"><i data-v-89a61d08="" class="bi bi-binoculars"></i> Show Trackers Manager </button>
      </div>
    </div>
    <div data-v-89a61d08="" class="osim-tracker-card pb-2 pt-0 pe-2 ps-2 bg-dark">
      <table data-v-89a61d08="" class="trackers-table table table-striped table-info mb-0">
        <thead data-v-89a61d08="">
          <tr data-v-89a61d08="">
            <th data-v-89a61d08="">Bug ID</th>
            <th data-v-89a61d08="">Module</th>
            <th data-v-89a61d08="">Component</th>
            <th data-v-89a61d08="">Product Stream</th>
            <th data-v-89a61d08=""><span data-v-89a61d08="" class="align-bottom me-1">Status</span><button data-v-89a61d08="" id="status-filter" type="button" class="btn btn-sm border-0 p-0" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false"><i data-v-89a61d08="" class="bi bi-funnel" title=""></i></button>
              <ul data-v-89a61d08="" class="dropdown-menu" aria-labelledby="status-filter"></ul>
            </th>
            <th data-v-89a61d08="">Resolution</th>
            <th data-v-89a61d08=""> Created date <i data-v-89a61d08="" class="bi-caret-up-fill bi"></i></th>
            <th data-v-89a61d08=""> Updated date <i data-v-89a61d08="" class="opacity-0 bi-caret-up-fill bi"></i></th>
          </tr>
        </thead>
        <tbody data-v-89a61d08=""></tbody>
      </table>
      <div data-v-89a61d08=""> No trackers to display </div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`flawTrackers > Correctly renders the component when there are trackers to display 1`] = `
"<div data-v-89a61d08="">
  <div data-v-89a61d08="" class="affects-trackers">
    <div data-v-89a61d08="" class="bg-dark rounded-top-2 text-info">
      <div data-v-89a61d08="" class="affect-trackers-heading d-flex p-1 pt-2 px-3">
        <h5 data-v-89a61d08="" class="m-0">Trackers</h5>
      </div>
      <!--v-if--><input data-v-89a61d08="" type="text" class="form-control border border-info focus-ring m-2 mb-1 p-1" style="width: 35ch;" placeholder="Filter Modules/Components...">
      <div data-v-89a61d08="" class="trackers-toolbar p-2 pt-1">
        <div data-v-89a61d08="" class="tracker-badges">
          <div data-v-89a61d08="" class="per-page-btn btn btn-sm badge"><i data-v-89a61d08="" style="pointer-events: auto;" class="bi bi-dash-square fs-6 my-auto" title="Reduce trackers per page"></i><span data-v-89a61d08="" class="mx-2 my-auto">Per page:</span><input data-v-89a61d08="" type="text" class="form-control mx-2" value="10"><i data-v-89a61d08="" style="pointer-events: auto;" class="bi bi-plus-square fs-6 my-auto" title="Increase trackers per page"></i></div>
          <!--v-if-->
        </div><button data-v-89a61d08="" type="button" class="btn btn-sm btn-info ms-auto"><i data-v-89a61d08="" class="bi bi-binoculars"></i> Show Trackers Manager </button>
      </div>
    </div>
    <div data-v-89a61d08="" class="osim-tracker-card pb-2 pt-0 pe-2 ps-2 bg-dark">
      <table data-v-89a61d08="" class="trackers-table table table-striped table-info mb-0">
        <thead data-v-89a61d08="">
          <tr data-v-89a61d08="">
            <th data-v-89a61d08="">Bug ID</th>
            <th data-v-89a61d08="">Module</th>
            <th data-v-89a61d08="">Component</th>
            <th data-v-89a61d08="">Product Stream</th>
            <th data-v-89a61d08=""><span data-v-89a61d08="" class="align-bottom me-1">Status</span><button data-v-89a61d08="" id="status-filter" type="button" class="btn btn-sm border-0 p-0" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false"><i data-v-89a61d08="" class="bi bi-funnel" title=""></i></button>
              <ul data-v-89a61d08="" class="dropdown-menu" aria-labelledby="status-filter">
                <li data-v-89a61d08=""><a data-v-89a61d08="" href="#" class="btn py-0 dropdown-item"><input data-v-89a61d08="" type="checkbox" class="form-check-input me-2"><span data-v-89a61d08="">NEW</span></a></li>
                <li data-v-89a61d08=""><a data-v-89a61d08="" href="#" class="btn py-0 dropdown-item"><input data-v-89a61d08="" type="checkbox" class="form-check-input me-2"><span data-v-89a61d08="">CLOSED</span></a></li>
              </ul>
            </th>
            <th data-v-89a61d08="">Resolution</th>
            <th data-v-89a61d08=""> Created date <i data-v-89a61d08="" class="bi-caret-up-fill bi"></i></th>
            <th data-v-89a61d08=""> Updated date <i data-v-89a61d08="" class="opacity-0 bi-caret-up-fill bi"></i></th>
          </tr>
        </thead>
        <tbody data-v-89a61d08="">
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0006" target="_blank">XXXX-0006 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-5">openshift-5</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-006">xxxx-0-006</td>
            <td data-v-89a61d08="" colspan="1">CLOSED</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-07-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-07-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0005" target="_blank">XXXX-0005 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-5">openshift-5</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-005">xxxx-0-005</td>
            <td data-v-89a61d08="" colspan="1">NEW</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-08-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-08-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0004" target="_blank">XXXX-0004 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-004">xxxx-0-004</td>
            <td data-v-89a61d08="" colspan="1">CLOSED</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-09-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-09-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0003" target="_blank">XXXX-0003 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-003">xxxx-0-003</td>
            <td data-v-89a61d08="" colspan="1">NEW</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-10-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-10-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0002" target="_blank">XXXX-0002 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-002">xxxx-0-002</td>
            <td data-v-89a61d08="" colspan="1">CLOSED</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-11-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-11-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0001" target="_blank">XXXX-0001 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-001">xxxx-0-001</td>
            <td data-v-89a61d08="" colspan="1">NEW</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-12-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-12-01 13:06 UTC</td>
          </tr>
        </tbody>
      </table>
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`flawTrackers > Displays embedded trackers manager 1`] = `
"<div data-v-89a61d08="">
  <div data-v-89a61d08="" class="affects-trackers">
    <div data-v-89a61d08="" class="bg-dark rounded-top-2 text-info">
      <div data-v-89a61d08="" class="affect-trackers-heading d-flex p-1 pt-2 px-3">
        <h5 data-v-89a61d08="" class="m-0">Trackers</h5>
      </div>
      <!--v-if--><input data-v-89a61d08="" type="text" class="form-control border border-info focus-ring m-2 mb-1 p-1" style="width: 35ch;" placeholder="Filter Modules/Components...">
      <div data-v-89a61d08="" class="trackers-toolbar p-2 pt-1">
        <div data-v-89a61d08="" class="tracker-badges">
          <div data-v-89a61d08="" class="per-page-btn btn btn-sm badge"><i data-v-89a61d08="" style="pointer-events: auto;" class="bi bi-dash-square fs-6 my-auto" title="Reduce trackers per page"></i><span data-v-89a61d08="" class="mx-2 my-auto">Per page:</span><input data-v-89a61d08="" type="text" class="form-control mx-2" value="10"><i data-v-89a61d08="" style="pointer-events: auto;" class="bi bi-plus-square fs-6 my-auto" title="Increase trackers per page"></i></div>
          <!--v-if-->
        </div><button data-v-89a61d08="" type="button" class="btn btn-sm btn-info ms-auto"><i data-v-89a61d08="" class="bi bi-binoculars"></i> Show Trackers Manager </button>
      </div>
    </div>
    <div data-v-89a61d08="" class="osim-tracker-card pb-2 pt-0 pe-2 ps-2 bg-dark">
      <table data-v-89a61d08="" class="trackers-table table table-striped table-info mb-0">
        <thead data-v-89a61d08="">
          <tr data-v-89a61d08="">
            <th data-v-89a61d08="">Bug ID</th>
            <th data-v-89a61d08="">Module</th>
            <th data-v-89a61d08="">Component</th>
            <th data-v-89a61d08="">Product Stream</th>
            <th data-v-89a61d08=""><span data-v-89a61d08="" class="align-bottom me-1">Status</span><button data-v-89a61d08="" id="status-filter" type="button" class="btn btn-sm border-0 p-0" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false"><i data-v-89a61d08="" class="bi bi-funnel" title=""></i></button>
              <ul data-v-89a61d08="" class="dropdown-menu" aria-labelledby="status-filter">
                <li data-v-89a61d08=""><a data-v-89a61d08="" href="#" class="btn py-0 dropdown-item"><input data-v-89a61d08="" type="checkbox" class="form-check-input me-2"><span data-v-89a61d08="">NEW</span></a></li>
                <li data-v-89a61d08=""><a data-v-89a61d08="" href="#" class="btn py-0 dropdown-item"><input data-v-89a61d08="" type="checkbox" class="form-check-input me-2"><span data-v-89a61d08="">CLOSED</span></a></li>
              </ul>
            </th>
            <th data-v-89a61d08="">Resolution</th>
            <th data-v-89a61d08=""> Created date <i data-v-89a61d08="" class="bi-caret-up-fill bi"></i></th>
            <th data-v-89a61d08=""> Updated date <i data-v-89a61d08="" class="opacity-0 bi-caret-up-fill bi"></i></th>
          </tr>
        </thead>
        <tbody data-v-89a61d08="">
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0006" target="_blank">XXXX-0006 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-5">openshift-5</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-006">xxxx-0-006</td>
            <td data-v-89a61d08="" colspan="1">CLOSED</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-07-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-07-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0005" target="_blank">XXXX-0005 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-5">openshift-5</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-005">xxxx-0-005</td>
            <td data-v-89a61d08="" colspan="1">NEW</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-08-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-08-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0004" target="_blank">XXXX-0004 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-004">xxxx-0-004</td>
            <td data-v-89a61d08="" colspan="1">CLOSED</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-09-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-09-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0003" target="_blank">XXXX-0003 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-003">xxxx-0-003</td>
            <td data-v-89a61d08="" colspan="1">NEW</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-10-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-10-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0002" target="_blank">XXXX-0002 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-002">xxxx-0-002</td>
            <td data-v-89a61d08="" colspan="1">CLOSED</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-11-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-11-01 13:06 UTC</td>
          </tr>
          <tr data-v-89a61d08="">
            <td data-v-89a61d08=""><a data-v-89a61d08="" href="http://jira-service:8002/browse/XXXX-0001" target="_blank">XXXX-0001 <i data-v-89a61d08="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-89a61d08="" colspan="1" title="openshift-1">openshift-1</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1" title="xxxx-0-001">xxxx-0-001</td>
            <td data-v-89a61d08="" colspan="1">NEW</td>
            <td data-v-89a61d08="" colspan="1"></td>
            <td data-v-89a61d08="" colspan="1">2024-12-01 13:05 UTC</td>
            <td data-v-89a61d08="" colspan="1">2024-12-01 13:06 UTC</td>
          </tr>
        </tbody>
      </table>
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
</div>"
`;
